src/backend/executor/nodeIndexonlyscan.c

```diff
@@ -87,33 +87,16 @@ IndexOnlyNext(IndexOnlyScanState *node)
 
 	if (scandesc == NULL)
 	{
-		if (node->ss.ps.plan->parallel_aware && estate->useMppParallelMode)
-		{
-			ParallelIndexScanDesc piscan;
-			ParallelEntryTag tag;
-			int localSliceId = LocallyExecutingSliceIndex(estate);
```

Comment:
Shall we use `localSliceId`, or remote it  when insert entry?

Reply:
GpFetchParallelDSMEntry has been removed , so  nothing to do with localSliceId.

Reply:
Shall we remove it when insert entry,  in GpInsertParallelDSMHash()?

Reply:
A little confused, insert entry needs a localSlicdId, but we don't use it when found entry?
How do nodes find their own dsm entry?

Reply:
you can found it in https://github.com/cloudberrydb/cloudberrydb/blob/main/src/backend/access/transam/parallel.c#L1898

Reply:
> No, we need to distinguish which is leader or worker.

Isn't it decided by lock `GpParallelDSMHashLock`?  Who get the lock is the first one.

Reply:
yes, you mentioned when remove entry, that is in GpDestroyParallelDSMEntry.

Reply:
I looked at these codes, It seems localSliceId is used to insert and destroy entry when execute plan.
We used to fetch dsm entry with localSliceId, but this pr sets parallel scandesc as other PG nodes after we insert/fetch an entry. And we use it to destroy that entry when plan ends.
The localSliceId is used to distinguish slices within a whole plan of  a session, it has nothing with leader or worker.

Do I understand right?

Reply:
right . localSliceId has nothing with leader or worker.